hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -602,6 +606,7 @@ func reconcileTokenSecret(tokenSecret *corev1.Secret, nodePool *hyperv1.NodePool
 
 	if tokenSecret.Data == nil {
 		tokenSecret.Data = map[string][]byte{}
+		tokenSecret.Annotations[TokenSecretLastUpdatedTokenIDAnnotation] = time.Now().Format(time.RFC3339Nano)
```

Comment:
I believe we need to store the rotated token in the secret to be able to recover the state if an ignition server restarts

Reply:
why do we need to "recover" the state though? If a restart happens the cache gets built with current token IDs which would get rotated after TTL/2 having an overlapping again preventing races. What problem does persisting the rotated token solve?

Reply:
So let's say in this case I have a inflight provision same way before at 1 second that uses the old token. If there's no way to rebuild this after a restart: if the ignition server pods restart for any reason (node reboots, etc) that then leads to provision failures.

It is a hard requriement for us to not have cases that can result in provisioning problems.

Reply:
So the steps would be just as before with the additional step of: restart ignition server pods before performing curl (do a pod delete of them `kubectl delete pods -l app=ignition-server`